src/apps/lwaftr/binding_table.lua

```diff
@@ -32,33 +32,72 @@ end
 
 local machine_friendly_binding_table
 
--- b4_v6 is for the B4, br_v6 is for the border router (lwAFTR)
+--[[
+b4_v6 is for the B4, br_v6 is for the border router (lwAFTR)
+
+Binding table format:
+   - Entry: {b4_v6, v4, { psid_info }, br_v6}
+   - PSID Info: {reserved_ports_bit_count=bits, psid_len=bits, shift=bits}
+
+Where:
+
+   - reserved_ports_bit_count = a bits.
+   - psid_len = k bits.
+   - shift = m bits.
+
+                     0                   1
+                     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5
+                    +-----------+-----------+-------+
+      Ports in      |     A     |    PSID   |   j   |
+   the CE port set  |    > 0    |           |       |
+                    +-----------+-----------+-------+
+                    |  a bits   |  k bits   |m bits |
+
+            Figure 2: Structure of a Port-Restricted Port Field
+
+Source: http://tools.ietf.org/html/rfc7597#section-5.1 
+--]]
+
 local function pton_binding_table(bt)
    local pbt = {}
+   local inserted = {}
    for _, v in ipairs(bt) do
-      local b4_v6 = ipv6:pton(v[1])
-      local pv4 = lwutil.rd32(ipv4:pton(v[2]))
+      local b4_v6, pv4, psid_info, br_v6 = unpack(v)
+      local reserved_ports_bit_count = psid_info.reserved_ports_bit_count or 0
+      local psid_len = psid_info.psid_len or 0
```

Comment:
I think instead of specifying reserved_ports_bit_count and psid_len, we should specify psid_len and shift.  WDYT?